server/api/controllers/locker-signout-controller.ts

```diff
@@ -0,0 +1,145 @@
+import { Request, Response } from "express";
+import { Logger } from "../../util/logger";
+// import { ReadWriteAPIController } from "./read-write-api-controller";
+import { ReadWriteController } from "./read-write-controller";
+import LockerData from "../../data/_hidden/locker-list.json";
+import { Locker } from "../../types/locker";
+import fs from "fs";
+
+export class LockerSignoutController {
+    static logger = new Logger("Locker Signout");
+
+    private static updateLockerStatus(term: string, userId: number, lockerNumber: number): boolean {
+      const lockerData = this.readLockerData();
+  
+      const targetTerm = lockerData[term];
+      const targetLocker = targetTerm.find((locker: Locker) => locker.lockerNumber === lockerNumber);
+  
+      if (targetLocker && targetLocker.userId === 0) {
+        targetLocker.userId = userId;
+        this.writeLockerData(lockerData); 
+        return true;
+      }
+  
+      return false;
+    }
+  
+    private static readLockerData(): { [key: string]: Locker[] } {
+      const jsonData = fs.readFileSync("server/data/_hidden/locker-list.json", "utf-8");
+      return JSON.parse(jsonData);
+    }
+  
+    private static writeLockerData(lockerData: { [key: string]: Locker[] }): void {
+      const jsonData = JSON.stringify(lockerData, null, 2);
+      fs.writeFileSync("server/data/_hidden/locker-list.json", jsonData, "utf-8");
+    }
+
+    private static generateJson(lockerData: { [key: string]: Locker[] }): string {
+      return JSON.stringify(lockerData, null, 2);
+    }
+    
+    /*
+      Handles the POST /api/locker-signout/request endpoint
+      - requires: userId
+      - checks if a locker is available
+      - if available, assigns the locker to the given user
+    */
+    static requestLocker(req: Request, res: Response) : void {
+      const { userId } = req.body;
+      const term = "placeholderTerm1"; // will set current term later
+  
+      const availableLocker = LockerData[term].find((locker: Locker) => locker.userId == 0);
+      if (!availableLocker) {
+        res.status(409).json({ error: "No available lockers" });
+        return;
+      }
+      const url = "_hidden/locker-list.json"
+      availableLocker.userId = userId;
+      ReadWriteController.overwriteJSONDataPath(
+        url,
+        () => {
+          return;
+        },
+        // need to generate json file -- currently only overwriting locally
```

Comment:
not sure if this comment is still meant to be here - this endpoint appears to be working!